azure-arc-for-servers/get-kvsecrets-from-arc-servers.ps1

```diff
@@ -0,0 +1,65 @@
+# run the following on your Azure ARC machines to get secrets from you Key Vault
+function Get-AccessToken {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $audience,
+        [string] $apiVersion = "2019-08-15"
+    )
+
+    $azcmagent = azcmagent show
+    if (-not ($azcmagent -like "*: Connected"))
+    {
+        Write-Error "Ensure Azure ARC machine is onboarded and himds sevice is running."
+        return
+    }
+        
+    $identityEndpoint = $env:IDENTITY_ENDPOINT
+    if(-not ($identityEndpoint)) {
+        $identityEndpoint = "http://localhost:40342/metadata/identity/oauth2/token"
+    }
+
+    $secretFile = $null
+    $endpointUri = "{0}?resource={1}&api-version={2}" -f $identityEndpoint, $audience, $apiVersion
+
+    $response = Invoke-WebRequest -Method GET -Uri $endpointUri `
+     -Headers @{Metadata="True"} -UseBasicParsing -SkipHttpErrorCheck
+    $challenge = $response.Headers["WWW-Authenticate"]
+    if ($challenge -match "Basic realm=.+")
+    {
+        $secretFile = ($challenge -split "Basic realm=")[1]
+    }
+
+    if (!$secretFile)
+    {
+        throw "something went wrong during requesting to identity service. Resposne content: $response.Content" 
+    }
+
+    $secret = Get-Content -Raw $secretFile
+    $response = Invoke-WebRequest -Method GET -Uri $endpointUri -Headers @{Metadata="True"; Authorization="Basic $secret"} -UseBasicParsing
```

Comment:
Why are you not using `Invoke-RestMethod`? You get JSON parsing for free. No need to convert the results.

Also, `-UseBasicParsing` should be removed.

Reply:
This is because I need to check the status code. Our service challenges the client by responding 401.  removed -UseBasicParsing